llpc/test/shaderdb/PipelineVsFs_TestNonRootPushConst.pipe

```diff
@@ -0,0 +1,171 @@
+
+
+; BEGIN_SHADERTEST
+; RUN: amdllpc -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: {{^// LLPC}} SPIRV-to-LLVM translation results
+; SHADERTEST: AMDLLPC SUCCESS
```

Comment:
This does not match any compilation results (LLVM IR, GCN, Pal metadata, etc.). How can we tell if the code does what its supposed to?

Reply:
It will crash when compiling this pipeline without this change.  This pipeline is aim to verify this change, not to special instructions. It is okay if it can be compiled successfully. Thanks.

Reply:
Can you add a comment with this explanation?

Reply:
I added it. Thanks.

Reply:
I don't see the checks that Steven suggested in the test, so I'm marking this as not resolved yet.

Reply:
Currently, are there such three test cases with compact mode?

Reply:
I don't know if they currently exist or not.  I believe the test that you have here would normally not spill the push constant, but you can look into that.  If it does spill you can probably change the type of mvp to float, and it will unspill it.  I don't know if there is a case for the compute shaders, but it should be easy enough to write.  I wrote a test for relocatable shaders that checks that it works when the push const is spilled: 

https://github.com/GPUOpen-Drivers/llpc/blob/b780577b54bcccb85c191d022b63ea5ce9cc72a4/llpc/test/shaderdb/relocatable_shaders/PipelineCs_PushConst.pipe.